### PR TITLE
handle continue on sync failure

### PR DESCRIPTION
**Pull request by Ishankoradia**
## Summary by CodeRabbit

* **New Features**
  * Configurable task execution strategy via a continue_on_sync_failure flag: default sequential fail-fast mode, or sync-tolerant mode that runs all data syncs first, collects/report sync errors, then proceeds to transforms.

* **Tests**
  * Added comprehensive tests covering sync detection, sequential execution, sync-tolerant behavior, failure propagation, and relevant edge cases.

[Review Change Stack](https://app.coderabbit.ai/change-stack/DalgoT4D/prefect-proxy/pull/228)

### CODE REVIEW

File: proxy/prefect_flows.py [hunk 1]
```diff
@@ -334,39 +371,51 @@ def deployment_schedule_flow_v4(
...
-                    run_airbyte_connection_flow_v1(task_config)
-
-                elif task_config["slug"] == "airbyte-clear":
-                    run_airbyte_conn_clear(task_config)
+def _run_tasks_sequentially(tasks: list):
+    """Original behavior: run all tasks sequentially, fail fast on any error"""
+    try:
+        for task_config in tasks:
+            _run_task(task_config)
+            sleep(30)
```
Comment (siddhant3030) [inline proxy/prefect_flows.py:369]: Here 

 Sync A → run → sleep(30)   ← useful, gives buffer before Sync B
  Sync B → run → sleep(30)   ← useful, gives buffer before Sync C
  Sync C → run → sleep(30)   ← pointless, nothing comes next in the loop

every pipeline run pays an extra 30s for no reason.

Comment (siddhant3030) [reply proxy/prefect_flows.py:369]: to fix adds enumerate to track which iteration we're on, and skips the sleep on the last one?
